test/functional/normal/langmap_spec.lua

```diff
@@ -0,0 +1,234 @@
+    --     LANGMAP_ADJUST'ed or not.
+    --     We could tell vgetorpeek() this information by making all
+    --     character-getting functions take another argument.
+    --     Alternatively we could tell vgetorpeek() by storing information in
+    --     the global state.
+    feed('qiiq')
+    eq(eval('@w'), 'w')
+  end)
+  it('conversions of mappings are recorded in macros', function()
+    -- This is a reasonably easy fix, but it doesn't make sense to fix this and
+    -- not translating text recorded in the macro when mappings are not
+    -- applied.
+    feed_command('nnoremap w l')
+    feed('qxiq')
+    eq(eval('@x'), 'w')
+  end)
+  -- These used to be exceptions, but with the new implementation they aren't
+  -- any more.
+  -- Because turning them back into exceptions requires modifying global state
+  -- and I think they shouldn't be exceptions anyway, I'm leaving them as they
+  -- are.
+  it(':s///c confirmation', function()
+    feed_command('set langmap=yn,ny')
+    feed('qa')
+    feed_command('s/i/w/gc')
+    feed('yynq')
+    expect('iiw www')
+    feed('u@a')
+    expect('iiw www')
+    eq(eval('@a'), ':s/i/w/gc\rnny')
+  end)
+  it('ask yes/no after backwards range', function()
+    feed_command('set langmap=yn,ny')
+    feed('dd')
+    insert([[
+    hello
+    there
+    these
+    are
+    some
+    lines
+    ]])
+    feed_command('4,2d')
+    feed('y')
+    expect([[
+    hello
+    there
+    these
+    are
+    some
+    lines
+    ]])
+  end)
+  describe('exceptions', function()
+    -- All "command characters" that 'langmap' does not apply to.
+    -- These tests consist of those places where some subset of ASCII
+    -- characters define certain commands, yet 'langmap' is not applied to
+    -- them.
+    -- n.b. I think these shouldn't be exceptions.
+    --      "Fixing" them is reasonably easy, but in case others don't like the
+    --      idea I'm not going to.
+    it('insert-mode CTRL-G', function()
+      feed_command('set langmap=jk,kj', 'd')
+      insert([[
+      hello
+      hello
+      hello]])
+      expect([[
+      hello
+      hello
+      hello]])
+      feed('qa')
+      feed('gg3|ahello<C-G>jx<esc>')
+      feed('q')
+      expect([[
+      helhellolo
+      helxlo
+      hello]])
+      eq(eval('@a'), 'gg3|ahellojx')
+    end)
+    it('command-line CTRL-\\', function()
+      feed_command('set langmap=en,ne')
+      feed(':<C-\\>e\'hello\'\r<C-B>put ="<C-E>"<CR>')
+      expect([[
+      iii www
+      hello]])
+    end)
+    it('command-line CTRL-R', function()
+      helpers.source([[
+        let i_value = 0
+        let j_value = 0
+        call setreg('i', 'i_value')
+        call setreg('j', 'j_value')
+        set langmap=ij,ji
+      ]])
+      feed(':let <C-R>i=1<CR>')
+      eq(eval('i_value'), 1)
+      eq(eval('j_value'), 0)
+    end)
+    -- pending('-- More -- prompt', function()
+    --   -- The 'b' 'j' 'd' 'f' commands at the -- More -- prompt
+    -- end)
```

Comment:
The test should be implemented, but I couldn't immediately see how to write it and just left it as a comment.